flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/operation/OperationExecutor.java

```diff
@@ -270,14 +270,21 @@ public Set<String> listDatabases(String catalogName) {
     public Set<TableInfo> listTables(
             String catalogName, String databaseName, Set<TableKind> tableKinds) {
         checkArgument(
-                Arrays.asList(TableKind.TABLE, TableKind.VIEW).containsAll(tableKinds),
+                Arrays.asList(TableKind.TABLE, TableKind.VIEW, TableKind.MATERIALIZED_TABLE)
```

Comment:
After some rethink, I think we'd better revert this logic, there's a bit of a problem with the implementation here. Since we haven't currently figured out the relationship between table and materialized table, it's better not to support this feature. We need a separate task to do this.

Reply:
Indeed, when I was writing this part, I had some hesitation. +1 revert this